src/lib/mina.ts

```diff
@@ -773,3 +805,143 @@ function defaultNetworkState(): NetworkValue {
     nextEpochData: cloneCircuitValue(epochData),
   };
 }
+
+async function verifyAccountUpdate(
+  account: LedgerAccount,
+  accountUpdate: AccountUpdate,
+  transactionCommitments: {
+    commitment: Field;
+    fullCommitment: Field;
+  },
+  proofsEnabled: boolean
+): Promise<void> {
+  let perm = account.permissions;
+
+  let { commitment, fullCommitment } = transactionCommitments;
+
+  // we are essentially only checking if the update is empty or an actual update
+  function includesChange(val: any): boolean {
+    if (val?.constructor === Array) {
+      return !(val as Array<any>).every((v) => v == null);
+    } else {
+      return val != null;
+    }
+  }
+
+  function permissionForUpdate(key: string): string {
+    switch (key) {
+      case 'appState':
+        return perm.editState;
+      case 'delegate':
+        return perm.setDelegate;
+      case 'verificationKey':
+        return perm.setVerificationKey;
+      case 'permissions':
+        return perm.setPermissions;
+      case 'zkappUri':
+        return perm.setZkappUri;
+      case 'tokenSymbol':
+        return perm.setTokenSymbol;
+      case 'timing':
+        return 'None'; // TODO: do timing?
+      case 'votingFor':
+        return perm.setVotingFor; // TODO: do timing?
+      case 'sequenceEvents':
+        return perm.editSequenceState;
+      case 'incrementNonce':
+        return perm.incrementNonce;
+      case 'send':
+        return perm.send;
+      case 'receive':
+        return perm.receive;
+      default:
+        return '';
+    }
+  }
+
+  const update = accountUpdate.toJSON().body.update;
+
+  let isValidProof = false;
+  let isValidSignature = false;
+
+  // we don't check if proofs aren't enabled
+  if (!proofsEnabled) isValidProof = true;
+
+  if (accountUpdate.authorization.proof && proofsEnabled) {
+    let publicInput = accountUpdateToPublicInput(accountUpdate);
+    let publicInputFields = ZkappPublicInput.toFields(publicInput);
+
+    const proof = SmartContract.Proof().fromJSON({
+      maxProofsVerified: 2, // TODO: this can probably vary?
+      proof: accountUpdate.authorization.proof!,
+      publicInput: publicInputFields.map((f) => f.toString()),
+    });
+
+    try {
+      let verificationKey = account.zkapp?.verificationKey!;
+      isValidProof = await verify(proof, verificationKey);
+    } catch (error) {
```

Comment:
I'm generally against silencing errors like this. If at some point, a proof breaks `verify` for some reason, I want it to throw the actual error with the actual stack trace. You could accomplish that and still display the nice error message below by doing something like
```ts
err.message = additionalMessage + (err.message ? '\n\n' + err.message : '')
```